packages/liveblocks-node/src/utils.ts

```diff
@@ -42,11 +42,55 @@ export function normalizeStatusCode(statusCode: number): number {
   }
 }
 
+type QueryParams =
+  | Record<string, string | number | null | undefined>
+  | URLSearchParams;
+
 /**
- * Concatenates a path to a URL.
+ * Safely but conveniently build a URLSearchParams instance from a given
+ * dictionary of values. For example:
+ *
+ *   {
+ *     "foo": "bar+qux/baz",
+ *     "empty": "",
+ *     "n": 42,
+ *     "nope": undefined,
+ *     "alsonope": null,
+ *   }
+ *
+ * Will produce a value that will get serialized as
+ * `foo=bar%2Bqux%2Fbaz&empty=&n=42`.
+ *
+ * Notice how the number is converted to its string representation
+ * automatically and the `null`/`undefined` values simply don't end up in the
+ * URL.
  */
-export function urljoin(baseUrl: string | URL, path: string): string {
-  const url = new URL(baseUrl);
-  url.pathname = path;
+function toURLSearchParams(
+  params: Record<string, string | number | null | undefined>
+): URLSearchParams {
```

Comment:
In future (if we have the use case), I believe, we could also accept an array value and use `append` instead of `set` appropriately so that we are able set the same parameter multiple times. 

Example - construction of urls like: `https://liveblocks.io/123?groupId=1&groupId=2`

Reply:
> _“In future (if we have the use case), I believe, we could also accept an array value and use append instead of set appropriately so that we are able set the same parameter multiple times.”_

Yep! Feel free to update accordingly when there is the need for that!